plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1909,6 +1911,207 @@ read_only::get_table_rows_result read_only::get_table_rows( const read_only::get
 #pragma GCC diagnostic pop
 }
 
+// prefix 17bytes: status(1 byte) + table_name(8bytes) + index_name(8 bytes)
+void read_only::make_prefix(eosio::name table_name,  eosio::name index_name, uint8_t status, vector<char>& prefix)const {
+   EOS_ASSERT(prefix.size() == 2 * sizeof(uint64_t) + 1, chain::contract_table_query_exception, "Invalid prefix");
+
+   prefix[0] = static_cast<char>(status);
+   vector<char>  bin;
+   bin.reserve(sizeof(uint64_t));
+   convert_to_key(table_name.to_uint64_t(), bin);
+   for(int i = 0; i < sizeof(uint64_t); ++i) {
+      prefix[i + 1] = bin[i];
+   }
+
+   bin.clear();
+   convert_to_key(index_name.to_uint64_t(), bin);
+   int offset = sizeof(uint64_t) + 1;
+   for(int i = 0; i < sizeof(uint64_t); ++i) {
+      prefix[offset + i] = bin[i];
+   }
+}
+
+read_only::get_table_rows_result read_only::get_kv_table_rows( const read_only::get_kv_table_rows_params& p)const {
+   name database_id = chain::kvram_id;
+   auto& gp = db.get_global_properties();
+   auto& kv_config = gp.kv_configuration;
+   const chain::kv_database_config& limits = kv_config.kvram;
+
+   auto&database = db.db();
+   auto kv_context = chain::create_kv_chainbase_context(const_cast<chainbase::database&>(database), database_id, chain::name{0}, {}, limits);
```

Comment:
I could be wrong, but I thought we had added a way to get a non-const reference to `chainbase::database` then we wouldn't have to cast away the const.

Reply:
get_kv_tables_rows() needs to be revisited once rocksdb changes are merged to develop to enable rocksdb  kv_context.

Reply:
Can we add a warning here in the code, and an additional Jira ticket to make sure that work is captured?

Reply:
I added a TO DO comment and will make a JIRA ticket for it.